core/src/main/java/org/apache/calcite/sql2rel/ConvertToChecked.java

```diff
@@ -92,8 +98,7 @@ static class ConvertRexToChecked extends RexShuttle {
           result = call;
         }
         return builder.makeCast(call.getType(), result);
-      } else if (!SqlTypeName.EXACT_TYPES.contains(resultType)
-          || (resultType == SqlTypeName.DECIMAL)) {
```

Comment:
Is this a related change?

Reply:
+1

Reply:
turns out that the second condition in the OR can never be true since it is covered by the previous if.